app/src/main/java/eu/kanade/presentation/more/onboarding/PermissionStep.kt

```diff
@@ -157,4 +187,26 @@ internal class PermissionStep : OnboardingStep {
             colors = ListItemDefaults.colors(containerColor = Color.Transparent),
         )
     }
+
+    @Composable
+    private fun PermissionSwitch(
+        title: String,
+        subtitle: String,
+        granted: Boolean,
+        modifier: Modifier = Modifier,
+        onToggleChange: (Boolean) -> Unit,
+    ) {
+        ListItem(
+            modifier = modifier,
+            headlineContent = { Text(text = title) },
+            supportingContent = { Text(text = subtitle) },
+            trailingContent = {
+                Switch(
+                    checked = granted,
+                    onCheckedChange = onToggleChange,
+                )
+            },
+            colors = ListItemDefaults.colors(containerColor = Color.Transparent),
+        )
```

Comment:
Since this and `PermissionCheckbox` share properties we should have a base composable that has a widget parameter that you pass into `trailingContent`